### DISCUSSION TITLE
query execution slow, unable to enable extensions or query parquet

### DISCUSSION THREAD

**Discussion by thebuck25** [duckdb/pg_duckdb#181] [General]
No change in query performance directly against tables (Postgres: 2.1s, pg_duckdb: 2.1s, Mac M1 DuckDB: 0.025s) which leads me to believe it's falling back to the pgsql engine.  Also unable to query parquet files, enable extensions, etc:

```
CREATE EXTENSION pg_duckdb;

WARNING:  To actually execute queries using DuckDB you need to run "SET duckdb.execution TO true;"
CREATE EXTENSION

Query returned successfully in 108 msec.

SET duckdb.execution TO true;

SET

Query returned successfully in 63 msec.

SELECT duckdb.enable_extension('iceberg');

ERROR:  function duckdb.enable_extension(unknown) does not exist
LINE 1: SELECT duckdb.enable_extension('iceberg');
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts. 

SQL state: 42883
Character: 8

SELECT SUM('DISTANCE') AS total
FROM read_parquet('r2://1101503dfsadfb........../test.parquet)

ERROR:  a column definition list is required for functions returning "record"
LINE 2: FROM read_parquet('r2://1101503dfsadfb......83479.r2....
             ^ 

SQL state: 42601
Character: 38
```
Postgres 16.3 Citus Cluster (Stackgres Operator) 1 Coordinator, 1 Shard with replicas - running in an Azure AKS x64 node
Compiled from source using custom docker image for Stackgres' Ubuntu containers running patroni
(Stackgres uses an older kernel w/ glibc 2.28)

Build image w/ make output is available here: https://hub.docker.com/repository/docker/killinterpol/quackbuster/general.

Let me know if you need more info, if I can troubleshoot further, run any available tests or otherwise contribute...

**Comment by wuputah** [comment]
Pardon my brevity, I'm answering from my phone. Hopefully this is enough to get you started:

* duckdb.install_extension
* read_parquet needs an AS statement

**Comment by mike-luabase** [comment]
> No change in query performance directly against tables

@wuputah maybe your postgres table had an unsupported datatype?

> The following data types are supported: numeric, character, binary, date/time, boolean, uuid, json, and arrays.

**Comment by thebuck25** [comment]
@wuputah thanks - **install**_extension returns false regardless of the extension I try to install:

```
WARNING:  (duckdb_install_extension) HTTP Error: Failed to download extension "iceberg" at URL "http://extensions.duckdb.org/17d598fc44/linux_amd64/iceberg.duckdb_extension.gz" (HTTP 403)
Extension "iceberg" is an existing extension.

Are you using a development build? In this case, extensions might not (yet) be uploaded.

Successfully run. Total query runtime: 291 msec.
1 rows affected.
```

Also, do you want me to do a PR to fix the readme?  
`Enable the DuckDB Iceberg extension using SELECT duckdb.enable_extension('iceberg') and read Iceberg files with iceberg_scan.`

Was able to query parquet after adding the second AS (missed that in README and I've never needed it using httpfs) however GROUP BY year() fails still. Not sure if date functions are supported yet.

@mike-luabase - the table in question contains only - uuid, numeric, character varying (255), timestamp w/o time zone.  Lmk if you want my table def & query, or is the varchar the issue?

**Comment by wuputah** [comment]
The iceberg problem should be resolved by #182
